main.go

```diff
@@ -76,27 +105,73 @@ func cloneorgrepos(ctx context.Context, client *github.Client, org string) error
 	var orgclone sync.WaitGroup
 	//iterating through the repo array
 	for _, repo := range orgRepos {
+		executeclone(repo, "/tmp/repos/org/"+*repo.Name, &orgclone)
+	}
+	orgclone.Wait()
+	fmt.Println("")
+	return nil
+}
+
+func findTeamByName(ctx context.Context, client *github.Client, org string, teamName string) (*github.Team, error) {
+
+	listTeamsOpts := &github.ListOptions{
+		PerPage: 10,
+	}
+	Info("Listing teams...")
+	for {
+		teams, resp, err := client.Organizations.ListTeams(ctx, org, listTeamsOpts)
+		check(err)
+		//check the name here--try to avoid additional API calls if we've found the team
+		for _, team := range teams {
+			if *team.Name == teamName {
+				return team, nil
+			}
+		}
+		if resp.NextPage == 0 {
+			break
+		}
+		listTeamsOpts.Page = resp.NextPage
+	}
+	return nil, nil
+}
+
+func cloneTeamRepos(ctx context.Context, client *github.Client, org string, teamName string) error {
+
+	// var team *github.Team
+	team, error := findTeamByName(ctx, client, org, teamName)
+	if team != nil {
+		Info("Cloning the repositories of the team: " + *team.Name + "(" + strconv.Itoa(*team.ID) + ")")
+		var teamRepos []*github.Repository
+		listTeamRepoOpts := &github.ListOptions{
+			PerPage: 10,
+		}
 
-		switch *cloneForks {
-		case true:
-			//cloning everything
-			orgclone.Add(1)
-			fmt.Println(*repo.CloneURL)
-			//cloning the individual org repos
-			go gitclone(*repo.CloneURL, "/tmp/repos/org/"+*repo.Name, &orgclone)
-		case false:
-			//not cloning forks
-			if !*repo.Fork {
-				orgclone.Add(1)
-				fmt.Println(*repo.CloneURL)
-				//cloning the individual org repos
-				go gitclone(*repo.CloneURL, "/tmp/repos/org/"+*repo.Name, &orgclone)
+		Info("Listing team repositories...")
+		for {
+			repos, resp, err := client.Organizations.ListTeamRepos(ctx, *team.ID, listTeamRepoOpts)
+			check(err)
+			teamRepos = append(teamRepos, repos...) //adding to the repo array
+			if resp.NextPage == 0 {
+				break
 			}
+			listTeamRepoOpts.Page = resp.NextPage
 		}
 
+		var teamclone sync.WaitGroup
+		//iterating through the repo array
```

Comment:
`teamclone.Add(1)` ?

Reply:
`executeclone` takes a pointer to `teamclone` (`&teamclone` in the call) and makes the `wg.Add()` call for each thread. After calling `executeclone`, `cloneTeamRepos` calls `teamclone.Wait()`